ktor-server/ktor-server-core/jsAndWasmShared/src/io/ktor/server/engine/EmbeddedServer.jsAndWasmShared.kt

```diff
@@ -44,9 +44,7 @@ actual constructor(
 
     private val modules = rootConfig.modules
 
-    public actual fun start(wait: Boolean): EmbeddedServer<TEngine, TConfiguration> {
-        addShutdownHook { stop() }
-
+    private fun prepareToStart() {
```

Comment:
I wonder if this function should also be suspending, replacing `CoroutineScope(application.coroutineContext).launch {` with `launch(application.coroutineContext) {`

Reply:
It can't be suspend here, as it's used also in non-suspend `start` here.
Also, the implementation is mostly copied from https://github.com/ktorio/ktor/blob/1f3bc45d77e64abb546597536a72d5252caa70b0/ktor-server/ktor-server-core/posix/src/io/ktor/server/engine/EmbeddedServerNix.kt#L60 and I would like to cleanup such differences during preparation for wasm-wasi support later, so that there will be MORE shared code.
So far, I haven't thought a lot, how we can improve this for nix/wasm so that it will work fine for both suspend and blocking way of starting the server